File: backend/app/schemas/user_dashboard.py
from datetime import date
from typing import Any

from pydantic import BaseModel


class RecentFeedbackRead(BaseModel):
    exercise_date: date | None = None
    exercise_type: str | None = None
    rpe: float | None = None
    pain_score_after: int | None = None
    discomfort: list[str] = []
    completion_rate: float | None = None


class FeedbackTrendPoint(BaseModel):
    date: date
    rpe: float
    pain: int | None = None
    completion_rate: float


class HealthRadarItem(BaseModel):
    metric: str
    value: float
    max: float


class RiskRuleHitRead(BaseModel):
    rule_id: str | int | None = None
    rule_name: str | None = None
    field_path: str | None = None
    hit_value: str | int | float | bool | None = None
    threshold: str | int | float | bool | None = None
    action_label: str | None = None
    risk_level: str | None = None
    explanation: str | None = None


class UserDashboardRead(BaseModel):
    current_risk_level: str | None
    expert_review_status: str
    today_can_exercise: bool
    today_block_reason: str | None = None
    weekly_completion_rate: float
    profile_completion_rate: float
    current_stage_goals: list[str]
    recent_feedback: RecentFeedbackRead
    monitoring_reminders: list[str]
    prescription_id: int | None = None
    prescription_version: int | None = None
    next_reassessment_date: date | None = None
    streak_days: int
    weekly_target_hits: int
    plan_completion_trend: list[float]
    feedback_trend: list[FeedbackTrendPoint] = []
    health_radar: list[HealthRadarItem] = []
    abnormal_feedback_count: int
    review_status_label: str
    prescription_summary: dict[str, Any] | None = None
    risk_rule_hits: list[RiskRuleHitRead] = []

File: backups/prescription_test_fix_20260606_225258/frontend/src/api/userDashboard.ts
import { apiClient } from "./client";

export interface RiskRuleHitSummary {
  rule_id?: string | number | null;
  rule_name?: string | null;
  name?: string | null;
  field_path?: string | null;
  path?: string | null;
  hit_value?: string | number | boolean | null;
  value?: string | number | boolean | null;
  threshold?: string | number | boolean | null;
  action?: string | null;
  action_label?: string | null;
  risk_level?: string | null;
  explanation?: string | null;
}

export interface UserDashboardSummary {
  current_risk_level: string | null;
  expert_review_status: string;
  today_can_exercise: boolean;
  today_block_reason: string | null;
  weekly_completion_rate: number;
  profile_completion_rate?: number | null;
  current_stage_goals: string[];
  recent_feedback: {
    exercise_date: string | null;
    exercise_type: string | null;
    rpe: number | null;
    pain_score_after: number | null;
    discomfort: string[];
    completion_rate: number | null;
  };
  monitoring_reminders: string[];
  prescription_id: number | null;
  prescription_version: number | null;
  next_reassessment_date: string | null;
  streak_days: number;
  weekly_target_hits: number;
  plan_completion_trend: number[];
  feedback_trend: Array<{
    date: string;
    rpe: number;
    pain: number | null;
    completion_rate: number;
  }>;
  health_radar: Array<{
    metric: string;
    value: number;
    max: number;
  }>;
  abnormal_feedback_count: number;
  review_status_label: string;
  prescription_summary: {
    cluster_label?: string | null;
    fitt_vp?: Record<string, unknown> | null;
    safety_notice?: string | null;
    reassessment?: string | null;
    contraindications?: string[];
  } | null;
  risk_rule_hits?: RiskRuleHitSummary[];
}

export function getUserDashboard() {
  return apiClient.get<UserDashboardSummary>("/user/dashboard").then((response) => response.data);
}
